test_operations from 1.6.4.post2 fails with GEOS 3.8

## Expected behavior and actual behavior.

Installing `shapely` 1.6.4.post2 with GEOS 3.8.0, one of the tests from `pytest` fails:

```
=================================== FAILURES ===================================
______________________ OperationsTestCase.test_operations ______________________

self = <tests.test_operations.OperationsTestCase testMethod=test_operations>

    def test_operations(self):
        point = Point(0.0, 0.0)
    
        # General geometry
        self.assertEqual(point.area, 0.0)
        self.assertEqual(point.length, 0.0)
        self.assertAlmostEqual(point.distance(Point(-1.0, -1.0)),
                               1.4142135623730951)
    
        # Topology operations
    
        # Envelope
        self.assertIsInstance(point.envelope, Point)
    
        # Intersection
        self.assertIsInstance(point.intersection(Point(-1, -1)),
>                             GeometryCollection)
E       AssertionError: <shapely.geometry.point.Point object at 0x7fccdc7c59b0> is not an instance of <class 'shapely.geometry.collection.GeometryCollection'>

tests/test_operations.py:26: AssertionError
```

I added a print statement and `print(point.intersection(Point(-1, -1)))` produces
```
POINT EMPTY
```

Expected result: all tests pass. 

See https://github.com/conda-forge/shapely-feedstock/pull/58 for the failed builds on conda-forge.

## Steps to reproduce the problem.

```
# get the code
git clone git@github.com:Toblerity/Shapely.git
cd Shapely
git reset --hard 1.6.4.post2
# set up a conda environment with python and GEOS 3.8.0
conda config --add channels conda-forge
conda config --set channel_priority strict
conda create -n test python=3.7 geos=3.8.0
conda activate test
# Follow the directions in the README
pip install -r requirements-dev.txt
export GEOS_DIR=~/miniconda3/envs/test
pip install -e .
pytest tests
```

## Operating system

Ubuntu 18.04.3

## Shapely version and provenance

1.6.4.post2 from pypi

--- a/tests/test_operations.py
+++ b/tests/test_operations.py
@@ -22,8 +22,7 @@ class OperationsTestCase(unittest.TestCase):
         self.assertIsInstance(point.envelope, Point)
 
         # Intersection
-        self.assertIsInstance(point.intersection(Point(-1, -1)),
-                              GeometryCollection)
+        self.assertTrue(point.intersection(Point(-1, -1)).is_empty)
 
         # Buffer
         self.assertIsInstance(point.buffer(10.0), Polygon)

